ledfx/virtuals.py

```diff
@@ -267,17 +267,22 @@ def set_effect(self, effect):
             _LOGGER.warning(error)
             raise ValueError(error)
 
-        self.transition_frame_total = (
-            self.refresh_rate * self._config["transition_time"]
-        )
-        self.transition_frame_counter = 0
-
-        if self._active_effect is None:
-            self._transition_effect = DummyEffect(self.pixel_count)
+        if (
+            self._config["transition_mode"] != "None"
+            and self._config["transition_time"] > 0
+        ):
+            self.transition_frame_total = (
+                    self.refresh_rate * self._config["transition_time"]
+            )
+            self.transition_frame_counter = 0
 
+            if self._active_effect is None:
+                self._transition_effect = DummyEffect(self.pixel_count)
+            else:
+                self.clear_transition_effect()
+                self._transition_effect = self._active_effect
         else:
             self.clear_transition_effect()
```

Comment:
speculative clear transition just in case, function protects against "not active"